TESTING.md

```diff
@@ -0,0 +1,82 @@
+# Test Strategy
+
+The overall test strategy can be described as _minimalistic_, rather than extensive. The goal of the tests is to verify that the application is deployed properly, so we will verify the main application features. 
+
+Decide the testing scope carefully, keeping in mind the maintainability of the tests and the value they provide.
+The tests described here are _deployment_ tests since their goal is to verify that the software is correctly deployed with all the inherent features. Both functional and non-functional characteristics are evaluated in these tests, focusing on the installation aspect. 
+
+* The core asset features work properly when deployed
+* There are no regression bugs 
+
+Before writing any test scenario, understand the primary purpose of the chart. Take a look at [the documentation about the chart under test](https://github.com/bitnami/charts/tree/master/bitnami) as well as the [docker image documentation](https://github.com/bitnami?q=docker&type=all&language=&sort=). This will give you a solid base for creating valuable test scenarios.
+
+## Common test cases
+
+* Important files and folders are present in the docker image
+* CRUD
+* File upload
+* SMTP (if applicable)
+* Establish a connection 
+* Login/Logout 
+* Plugins (if applicable)
+
+## HOW-TO:
+
+### Tools 
+
+The current test system supports a set of tools that are listed below. Test execution is done on temporary environments which are deployed and destroyed on the fly. In order to trigger the test execution in these test environments, internal approval is needed. Keeping in mind the acceptance criteria as well as the best practices for test automation using the supported tool will help you get your PR approved. 
+
+#### Dynamic testing
+
+If your asset has a user interface, you will **need** to include the following tests:
+
+* [Cypress](https://docs.cypress.io/guides/overview/why-cypress) (functional tests)
+* [Goss](https://github.com/aelsabbahy/goss/blob/master/docs/manual.md) (infrastructure and configuration tests)
+* Health checks
+* Resilience check (needed for all assets that require high availability (eg. databases))
+
+#### Static analysis
+
+Static analysis is needed for all assets. 
+
+* [Trivy](https://github.com/aquasecurity/trivy)
+* Helm lint
+
+## Test acceptance criteria
+
+In order for your test code PR to be accepted the following criteria must be fulfilled. 
+
+### Generic
+
+- [ ] Minimum of 5 test cases per test type is needed 
+- [ ] Key features of the asset need to be covered 
+- [ ] Tests need to contain assertions 
+- [ ] Tests need to be stateless 
+- [ ] Tests need to be independent 
+- [ ] Tests need to be retry-able
+- [ ] Tests need to be executable in any order 
```

Comment:
I've seen situations where, in order to test one feature you need to create several elements prior to that. I remember the case of applications like `weblate`: I want to test the file upload, but in order to upload a file first I need to create a project, but in order to create a project I need to create a workspace. 

So, in order to test the file upload, should the test create the workspace and the project? If that's the case, is it worth creating separate tests for workspace and project creation?

Reply:
Each test should prepare its state and not rely on preconditions. This means that we will sometimes have repeated steps due to a need for prerequisites (ex. login will appear in many test cases), but the added benefit will be tests that are less flaky and can be executed in any order. 
This might mean that we use one test to verify multiple functionalities (create a workspace, create a project, file upload).